### PR TITLE
tx-bakery: separate ogmios launcher and client && use url to connect to ogmios service

### PR DESCRIPTION
**Pull request by chfanghr**

### CODE REVIEW

File: tx-bakery/src/utils/ogmios/launcher.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    Some(Url::parse(&format!("http://127.0.0.1:{}", config.port)).unwrap())
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        }
+        .ok_or(OgmiosError::StartupError(anyhow!(
+            "Unable to determine the url endpoint of the ogmios backend"
+        )))?;
```
Comment (bladyjoker) [inline tx-bakery/src/utils/ogmios/launcher.rs:131]: Why are you complicating this. 

You have the server part, which is the Launcher, and you have the Client part.

They are configured differently, no need to be smart and try to derive a client connection string given a server listening string.

Imo remove this part and just use the client part where you provide the URL.

File: tx-bakery/tests/main.rs [hunk 2]
```diff
@@ -770,15 +770,18 @@ mod tests {
     #[tokio::test]
     #[serial]
     async fn init_tx_bakery() -> Result<()> {
-        let (_plutip, ogmios) = setup_plutip_test().await;
-        TxBakery::init(&ogmios).await?;
+        let (_plutip, ogmios_launcher) = setup_plutip_test().await;
+        let ogmios = ogmios_launcher.connect().await.unwrap();
```
Comment (bladyjoker) [inline tx-bakery/tests/main.rs:774]: Well not really:

Launcher can do `launch()`, a Client can do `connect()`. Having a `connect()` on Launcher is what's making this complicated.

**Review by bladyjoker — COMMENTED** [review]
I'd say have two types OgmiosLauncher with a method `launch` and OgmiosClient with a method `connect`. Don't try to derive the parameters of OgmiosClient from OgmiosLauncher.

File: tx-bakery/src/utils/ogmios/client.rs [hunk 3]
```diff
@@ -1,42 +1,191 @@
...
+    }
+
+    pub async fn check_health(&self) -> Result<OgmiosHealth> {
+        Ok(reqwest::Client::new()
+            .get(self.get_restful_health_url())
+            .send()
+            .await?
+            .json::<OgmiosHealth>()
+            .await?)
+    }
```
Comment (bladyjoker) [inline tx-bakery/src/utils/ogmios/client.rs:93]: Why is this here and not in the actual OgmiosClient?

**Review by bladyjoker — APPROVED** [review]
Looks good! Small comment

File: tx-bakery/src/utils/ogmios/client.rs (hunk 3 above)
Comment (chfanghr) [reply tx-bakery/src/utils/ogmios/client.rs:93]: This is a restful request, not websocket

Comment (chfanghr) [reply tx-bakery/src/utils/ogmios/client.rs:93]: But yeah I see now, it makes sense to move the function
